### PR TITLE
Split providers into their own modules

### LINKED ISSUE #46: Split providers into their own packages
#### What would you like to be added:

Split each provider into it's own package e.g. salmon-provider-tiingo

#### Why is this needed:

Better code maintainability

### PR DESCRIPTION
**Pull request by monstrobishi**
#### What kind of PR is this?:

/kind feature

#### What this PR does / why we need it:

Split providers into their own modules

#### Which issue(s) does this PR fixes?:

Fixes #46

### CODE REVIEW

**Comment by fuxingloh** [comment]
Are we deploying each provider individually?

**Review by canonbrother — APPROVED** [review]

**Comment by monstrobishi** [comment]
@fuxingloh yes

**Review by jingyi2811 — APPROVED** [review]

**Comment by fuxingloh** [comment]
> @fuxingloh yes

I think you should do this instead, to have a more organic structure. Allowing further isolation of concerns.

```txt
salmon/packages/
├── salmon-core/
│   ├── src/index.ts
│   └── package.json for core oracle stuff
├── salmon-provider-finnhub/
│   ├── src/lambda.ts
│   ├── src/finnhub.ts
│   └── package.json with ncc
├── salmon-provider-iexcloud/
│   ├── src/lambda.ts
│   ├── src/iexcloud.ts
│   └── package.json with ncc

```

**Comment by monstrobishi** [comment]
@fuxingloh

I like this but I think it would introduce too many touch points when adding new providers. It would require modifying the github workflow and a few other places every time.

A reasonable compromise would be having a package for the lambda functions, with each file representing a provider, and ncc would package them all. Then the lambda function can specify which handler it wants

**Comment by monstrobishi** [comment]
Semantically for me it also doesn't make sense for the provider package to contain the "entry point" for the project. If you insist on this layout then I might as well go all the way and create a `salmon-lambda-tiingo` package alongside the provider package

File: packages/salmon-provider-tiingo/package.json [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    "dist"
+  ],
+  "scripts": {
+    "build": "tsc"
+  },
+  "peerDependencies": {
+    "bignumber.js": "^9.0.1"
+  },
+  "dependencies": {
+    "@defichain/jellyfish-json": "next",
```
Comment (fuxingloh) [inline packages/salmon-provider-tiingo/package.json:41]:
```suggestion
    "@defichain/salmon-price-functions": "0.0.0",
```

mono repo not properly configured?

File: packages/salmon-provider-iexcloud/package.json [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+  "types": "dist",
+  "files": [
+    "dist"
+  ],
+  "scripts": {
+    "build": "tsc"
+  },
+  "peerDependencies": {
+    "bignumber.js": "^9.0.1"
+  },
```
Comment (fuxingloh) [inline packages/salmon-provider-iexcloud/package.json:39]: should not need to provide peer depedendencies

**Review by fuxingloh — APPROVED** [review]
I checked the configs and the main `src/` in `salmon-lambda-functions`. I didn't check the tests again, as that will take hours again.

**Review by canonbrother — APPROVED** [review]

**Review by jingyi2811 — APPROVED** [review]

**Pull request merged**